### PR TITLE
Allow PHP 8.0

### LINKED ISSUE #142: PHP 8 support
PHP version 8 came out recently and I did not see any issues tracking support for it yet so I figured I would create one. The current version constraint prevents me from upgrading my application to it, so I would definitely appreciate if compatibility could be checked and the constraint could be updated accordingly.

### PR DESCRIPTION
**Pull request by iggyvolz**
Add PHP 8 to allowed version list for Restcord.  PHPUnit tests passed without issues, not sure how to run stupidFunctionalTest.  Fixes #142

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -3,12 +3,12 @@
     "description":  "REST Library for the Discord API",
     "type":         "library",
     "require":      {
-        "php":                        "^5.6|^7.0",
+        "php":                        "^5.6|^7.0|^8.0",
```
Comment (aequasi) [inline composer.json:6]: Lets fully shore up to minimum PHP EOL versions. Might as well

```suggestion
        "php":                        "^7.3|^8.0",
```

File: composer.json [hunk 2]
```diff
@@ -3,12 +3,12 @@
...
         "guzzlehttp/guzzle":          "^6.3",
         "guzzlehttp/guzzle-services": "^1.0",
         "monolog/monolog":            "^1.22|^2.0",
         "netresearch/jsonmapper":     "^1.4",
-        "symfony/options-resolver":   "^2.6|^3.0|^4.0|^5.0"
+        "symfony/options-resolver":   "^2.6|^3.0"
```
Comment (aequasi) [inline composer.json:11]: Not sure this is a good change

```suggestion
"symfony/options-resolver":   "^3.4|^4.4|^5.1"
```

File: composer.json [hunk 3]
```diff
@@ -29,11 +29,11 @@
...
     "require-dev":  {
         "beberlei/assert":          "^2.7",
-        "gossi/php-code-generator": "^0.4.1",
+        "gossi/php-code-generator": "^0.5.0",
         "symfony/console":          "^2.6|^3.0|^4.0",
         "symfony/process":          "^2.6|^3.0|^4.0",
         "symfony/var-dumper":       "^2.6|^3.0|^4.0",
         "twig/twig":                "^1.20|^2.0",
-        "phpunit/phpunit":          "^6.0|^5.0"
+        "phpunit/phpunit":          "^8.0"
```
Comment (aequasi) [inline composer.json:37]:
```suggestion
        "phpunit/phpunit":          "^8.0|^9.0"
```

File: .travis.yml [hunk 4]
```diff
@@ -9,6 +9,10 @@ matrix:
     include:
         - php: 7.0
         - php: 7.1
+        - php: 7.2
+        - php: 7.3
+        - php: 7.4
+        - php: 8.0
```
Comment (aequasi) [inline .travis.yml:15]: Remove 7.0 and 7.1

```suggestion
        - php: 7.3
        - php: 7.4
        - php: 8.0
```

File: composer.json (hunk 2 above)
Comment (iggyvolz) [reply composer.json:11]: I was getting weird errors from Composer with this line earlier - but this seems to work

**Comment by iggyvolz** [comment]
Am I doing something wrong with the CI?  integration/styleci/pr is being marked as misconfigured, and travis isn't running at all.

**Comment by aequasi** [comment]
Naw, i just need to fix it on my side

**Pull request merged**